python/GafferSceneTest/ParentTest.py

```diff
@@ -528,5 +528,127 @@ def testContextVariableForParent( self ) :
 				[ "/b/sphere" ]
 			)
 
+	def testParentContextVariable( self ) :
+
+		# Parent a sphere at `/a` and a grid at `/b`.
+
+		sphere = GafferScene.Sphere()
+		sphere["sets"].setValue( "set1" )
+
+		grid = GafferScene.Grid()
+
+		switch = Gaffer.NameSwitch()
+		switch.setup( sphere["out"] )
+		switch["selector"].setValue( "${parent}" )
+		switch["in"].resize( 3 )
+		switch["in"][1]["name"].setValue( "/a" )
+		switch["in"][1]["value"].setInput( sphere["out"] )
+		switch["in"][2]["name"].setValue( "/b" )
+		switch["in"][2]["value"].setInput( grid["out"] )
+
+		collect = GafferScene.CollectScenes()
+		collect["rootNames"].setValue( IECore.StringVectorData( [ "a", "b" ] ) )
+
+		filter = GafferScene.PathFilter()
+		filter["paths"].setValue( IECore.StringVectorData( [ "/a", "/b" ] ) )
+
+		parent = GafferScene.Parent()
+		parent["in"].setInput( collect["out"] )
+		parent["children"][0].setInput( switch["out"]["value"] )
+		parent["filter"].setInput( filter["out"] )
+		parent["parentVariable"].setValue( "parent" )
+
+		# Check the scene is as we expect
+
+		self.assertSceneValid( parent["out"] )
+
+		self.assertEqual( parent["out"].childNames( "/a" ), IECore.InternedStringVectorData( [ "sphere" ] ) )
+		self.assertEqual( parent["out"].childNames( "/a/sphere" ), IECore.InternedStringVectorData() )
+		self.assertEqual( parent["out"].childNames( "/b" ), IECore.InternedStringVectorData( [ "grid" ] ) )
+		self.assertEqual( parent["out"].childNames( "/b/grid" ), IECore.InternedStringVectorData( [ "gridLines", "centerLines", "borderLines" ] ) )
+
+		self.assertScenesEqual(
+			sphere["out"],
+			parent["out"],
+			scenePlug2PathPrefix = "/a"
+		)
+
+		self.assertPathHashesEqual(
+			sphere["out"], "/sphere",
+			parent["out"], "/a/sphere",
+		)
+
+		self.assertScenesEqual(
+			grid["out"],
+			parent["out"],
+			scenePlug2PathPrefix = "/b",
+			# Don't want to check sets, because the grid has no sets.
+			checks = self.allSceneChecks - { "sets" }
+		)
+
+		for path in [ "/grid", "/grid/centerLines", "/grid/gridLines", "/grid/borderLines" ] :
+			self.assertPathHashesEqual(
+				grid["out"], path,
+				parent["out"], "/b" + path,
+			)
+
+		# Rename the parent variable. This should dirty all the output plugs and make the NameSwitch
+		# output an empty scene.
+
+		cs = GafferTest.CapturingSlot( parent.plugDirtiedSignal() )
+		parent["parentVariable"].setValue( "x" )
+
+		self.assertLessEqual(
```

Comment:
What does `LessEqual` mean when applied to sets?  Should this just be `assertEqual`?

Reply:
It means that `a` is a subset of `b`. Not super obvious I know, and I did spend a bit of time looking for alternatives. Without going writing custom assertions, the next best is `assertTrue( a.issubset( b ) )`, but the error messages from that are much less meaningful because they don't include the values.

Reply:
I've added an explanatory comment in the code in e4b93d0.